ci/test_java.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+# Copyright (c) 2022-2025, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-logger "Downloading artifacts from previous jobs"
+CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
+
+rapids-logger "Generate Java testing dependencies"
+
+ENV_YAML_DIR="$(mktemp -d)"
+
+rapids-dependency-file-generator \
+  --output conda \
+  --file-key test_java \
+  --prepend-channel "${CPP_CHANNEL}" \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch)" | tee "${ENV_YAML_DIR}/env.yaml"
+
+rapids-mamba-retry env create --yes -f "${ENV_YAML_DIR}/env.yaml" -n test
+
+export CMAKE_GENERATOR=Ninja
+
+# Temporarily allow unbound variables for conda activation.
+set +u
+conda activate test
+set -u
+
+rapids-print-env
+
+rapids-logger "Check GPU usage"
+nvidia-smi
+
+EXITCODE=0
+trap "EXITCODE=1" ERR
+set +e
+
+rapids-logger "Run Java build and tests"
+
+bash ./build.sh java
```

Comment:
Will you eventually want to publish packages for the Java bindings, instead of doing a full rebuild from source here on a GPU runner? I hope so, as distributing packages would make it easier for users to integrate this code into their projects.

If yes, could you please link to a tracking issue for that work in a comment here, to make it clear that this `build.sh` should be reverted (in favor of installing packages) in the future?

Reply:
Thanks for the suggestion! Yes, we plan to separate Java build and test in CI. I’ve filed [#868](https://github.com/rapidsai/cuvs/issues/868) to track that.